### DISCUSSION TITLE
Numerical Instability - Flow in pipe with in pipe with nozzles - Evaporation and water deposition

### DISCUSSION THREAD

**Discussion by AlejandroEngESC** [firemodels/fds#13605] [Q&A] [answered]
Dear community, 
I'm currently running a numerical simulation using Pyrosim in order to first simulate the flow within a pipe of a heated Nafta flux (110ºC) in which a nozzle (Industrial nozzle predefined in Pyrosim) of liquid water is introduced (around 30ºC), with the intention of seing how both fluids flow through the pipe system. According with thee theory the liquid water, introduce by a modelized Pyrosim nozzle, should vaporize at least one part, creating 3 different fluid phases, nafta gas, liquid water and tIn the attached model you should see how I've modelled all fluids, introducing a supply for the nafta and then defining the water as a specie with initial fraction of gas set as zero and defining in liquid all the variables needed for the phase change (heat oif vaporization, specific heat, vaporization temperature...). 
As written in the subjet of this question, once the particles start to be injected in the model (nozzle activated after 20s) a numerical instability is achieved. The .fds file is attached hereafter, prepared with 2023.3.1206 Pyrosim version and run with 6.8.0 FDS version.  

FDS file:
&HEAD CHID='PyrosimModel'/
&TIME T_END=40.0/
&DUMP DT_RESTART=300.0, DT_SL3D=0.25/
&MISC GVEC=-1.0E-10,-1.0E-10,-9.81, TMPA=110.0/

&MESH ID='Mesh02-07-02', IJK=33,18,20, XB=30.15,31.14,-3.81203,-3.270677,7.716418,8.313433/
&MESH ID='Mesh02-11-02', IJK=34,18,20, XB=31.14,32.16,-3.81203,-3.270677,7.716418,8.313433/
&MESH ID='Mesh02-14-02', IJK=28,80,20, XB=32.16,33.0,-6.218045,-3.81203,7.716418,8.313433/
&MESH ID='Mesh02-15-02', IJK=28,18,20, XB=32.16,33.0,-3.81203,-3.270677,7.716418,8.313433/
&MESH ID='Mesh02-19-02', IJK=50,18,20, XB=33.0,34.5,-3.81203,-3.270677,7.716418,8.313433/
&MESH ID='Mesh02-23-02', IJK=50,18,20, XB=34.5,36.0,-3.81203,-3.270677,7.716418,8.313433/

&SPEC ID='N-PENTANE'/
&SPEC ID='ISO-OCTANE'/
&SPEC ID='N-HEPTANE'/
&SPEC ID='N-HEXANE'/
&SPEC ID='Water Drops', FYI='Customized Water Droplets', DIFFUSIVITY=2.5E-5, VISCOSITY=1.0E-3, SPECIFIC_HEAT=4.18, REFERENCE_TEMPERATURE=100.0, DENSITY_LIQUID=1000.0, SPECIFIC_HEAT_LIQUID=4.18, VAPORIZATION_TEMPERATURE=100.0, MELTING_TEMPERATURE=0.0, HEAT_OF_VAPORIZATION=2500.0, ENTHALPY_OF_FORMATION=-285.83/

&PART ID='Water',
      SPEC_ID='Water Drops',
      DIAMETER=10.0,
      MONODISPERSE=.TRUE.,
      AGE=60.0,
      SAMPLING_FACTOR=1,
      INITIAL_TEMPERATURE=35/

&PROP ID='Generic Industrial Spray',
      PART_ID='Water',
      PARTICLES_PER_SECOND=20000,
      FLOW_RATE=25.0,
      FLOW_RAMP='Generic Industrial Spray_FLOW_RAMP',
      ORIFICE_DIAMETER=3.0E-3,
      SPRAY_ANGLE=1.0,3.0/
&RAMP ID='Generic Industrial Spray_FLOW_RAMP', T=0.0, F=0.0/
&RAMP ID='Generic Industrial Spray_FLOW_RAMP', T=20.0, F=0.0/
&RAMP ID='Generic Industrial Spray_FLOW_RAMP', T=21.0, F=1.0/
&RAMP ID='Generic Industrial Spray_FLOW_RAMP', T=60.0, F=1.0/
&DEVC ID='NOZZLE', PROP_ID='Generic Industrial Spray', XYZ=32.575,-5.8,7.985, ORIENTATION=0.0,1.0,0.0, QUANTITY='TIME', SETPOINT=0.0/
&SURF ID='ADIABATIC',
      COLOR='GRAY 80',
      ADIABATIC=.TRUE./
&SURF ID='Supply',
      RGB=26,204,26,
      TMP_FRONT=110.0,
      MASS_FLUX=11.87248,10.41436,8.95585,7.4989,
      SPEC_ID='ISO-OCTANE','N-HEPTANE','N-HEXANE','N-PENTANE',
      TAU_MF=1.0,1.0,1.0,1.0/

Therefore some questions arise:
1- Is it well defined the species in order to achieve the theoretical evaporation of the water particles? If not, what are the changes to perform in order to be able to see the exact ammount of water converted to steam and where? 
2-Is it well defined withing the water particles "initial mass fraction" as zero? I belive that since you are imposing a mass fluz in the nozzle, and the particles set as liquid water, this "initial mass fraction" is set to define the ammount of gass that is injected at the nozzle inlet, ther
efore is set as zero, could you please confirm/correct? Since my intention is to inject just liquid water, with the possibility of vaporization in the model, is it correctly defined?
2- I've tested different models with and without global gravity defined. Once the gravity is set to a really small value (not zero, just to avoid software problems),  I see no changes regarding the behaviour of the particles. Therefore my question is, are there any other gravity parameters that only affect to the particles injected? 
3- As you will see in the attached model, the pipe is defined as a rectangular obstruction with a circular wall to define the pipe, with the intention of the mesh to avoid the rectangular wall and let the fluid only flow through it's interior. Is there any other alternative to define this exact model without this method? Using 1D ducts I belive is not the right way since one parameter that is necessary of this model to be extracted is the ammount of liquid water that reaches the elbow. 
4- Is there any alternative of defining the inflow fluid as Nafta without using the "supply" condition? 

Thank you in advance for your time and your fast repply,

**Comment by drjfloyd** [comment]
Suggest 
-just use WATER VAPOR as the species and not trying to define it. For example the gas specific heat of water vapor is more like 2 rather t
-use the current release
-make sure you read the FDS guides and understand how to set boundary conditions.  Right now your input is for a sealed box with no inflow of fuel vapor.

**Reply by AlejandroEngESC** [reply]
Hello and thank you for your fast reply, 

Regarding your first answer, in case I use water vapor as the specie instead of defining it, will it take into consideration the possibility of evaporation during the simulation? Or is there anything else I should include?
Regarding your third answer, at the moment I'm using a supply with different mass fluxes simulating something similar to nafta, and before the simulation breaks, I'm able to see how the fluid is entering the domain, in fact I've tried this configuration with a more simple geometry (just simple cubic mesh) and It worked fine with this definition of inlet, could you explain further why the fuel vapor is not entering the domain?

Thank you very much for your time

**Comment by drjfloyd** [comment]
How are sprinkler inputs done in the dozen or so examples in the FDS User and Verificaiton Guide?

Which line of input in your first post tells FDS where to inject your mixture of fuel gasses?

**Comment by AlejandroEngESC** [comment]
Hello, you were right, I just missed to copy the last part of my .fds file. Hereafter you will find the complete one, could you please advice whether is correct or not? I've already reviewed some Varification examples about nozzles and I modelled it accordingly, although I belive the problems are coming from the vaporization topic. Could you please answer to my last question "in case I use water vapor as the specie instead of defining it, will it take into consideration the possibility of evaporation during the simulation? Or is there anything else I should include?" I can't find the answer in the FDS guide (NIST Sixth Exition, April 18, 2023). 

The complete .fds file: 
TuberiaRepsol.fds
Generated by PyroSim 2023.3.1206
21 oct 2024, 10:09:34

&HEAD CHID='TuberiaRepsol'/
&TIME T_END=40.0/
&DUMP DT_RESTART=300.0, DT_SL3D=0.25/
&MISC GVEC=-1.0E-10,-1.0E-10,-9.81, TMPA=110.0/

&MESH ID='Mesh02-07-02', IJK=33,18,20, XB=30.15,31.14,-3.81203,-3.270677,7.716418,8.313433/
&MESH ID='Mesh02-11-02', IJK=34,18,20, XB=31.14,32.16,-3.81203,-3.270677,7.716418,8.313433/
&MESH ID='Mesh02-14-02', IJK=28,80,20, XB=32.16,33.0,-6.218045,-3.81203,7.716418,8.313433/
&MESH ID='Mesh02-15-02', IJK=28,18,20, XB=32.16,33.0,-3.81203,-3.270677,7.716418,8.313433/
&MESH ID='Mesh02-19-02', IJK=50,18,20, XB=33.0,34.5,-3.81203,-3.270677,7.716418,8.313433/
&MESH ID='Mesh02-23-02', IJK=50,18,20, XB=34.5,36.0,-3.81203,-3.270677,7.716418,8.313433/

&SPEC ID='N-PENTANE'/
&SPEC ID='ISO-OCTANE'/
&SPEC ID='N-HEPTANE'/
&SPEC ID='N-HEXANE'/
&SPEC ID='Water Drops', FYI='Customized Water Droplets', DIFFUSIVITY=2.5E-5, VISCOSITY=1.0E-3, SPECIFIC_HEAT=4.18, REFERENCE_TEMPERATURE=100.0, DENSITY_LIQUID=1000.0, SPECIFIC_HEAT_LIQUID=4.18, VAPORIZATION_TEMPERATURE=100.0, MELTING_TEMPERATURE=0.0, HEAT_OF_VAPORIZATION=2500.0, ENTHALPY_OF_FORMATION=-285.83/

&PART ID='Water',
      SPEC_ID='Water Drops',
      DIAMETER=10.0,
      MONODISPERSE=.TRUE.,
      AGE=60.0,
      SAMPLING_FACTOR=1,
      INITIAL_TEMPERATURE=35/

&PROP ID='Generic Industrial Spray',
      PART_ID='Water',
      PARTICLES_PER_SECOND=20000,
      FLOW_RATE=25.0,
      FLOW_RAMP='Generic Industrial Spray_FLOW_RAMP',
      ORIFICE_DIAMETER=3.0E-3,
      SPRAY_ANGLE=1.0,3.0/
&RAMP ID='Generic Industrial Spray_FLOW_RAMP', T=0.0, F=0.0/
&RAMP ID='Generic Industrial Spray_FLOW_RAMP', T=20.0, F=0.0/
&RAMP ID='Generic Industrial Spray_FLOW_RAMP', T=21.0, F=1.0/
&RAMP ID='Generic Industrial Spray_FLOW_RAMP', T=60.0, F=1.0/
&DEVC ID='NOZZLE', PROP_ID='Generic Industrial Spray', XYZ=32.575,-5.8,7.985, ORIENTATION=0.0,1.0,0.0, QUANTITY='TIME', SETPOINT=0.0/
&SURF ID='ADIABATIC',
      COLOR='GRAY 80',
      ADIABATIC=.TRUE./
&SURF ID='Supply',
      RGB=26,204,26,
      TMP_FRONT=110.0,
      MASS_FLUX=11.87248,10.41436,8.95585,7.4989,
      SPEC_ID='ISO-OCTANE','N-HEPTANE','N-HEXANE','N-PENTANE',
      TAU_MF=1.0,1.0,1.0,1.0/

[Mesh and Obstructions info not included]

&VENT ID='Mesh Vent: Mesh02-23-02 [XMAX]', SURF_ID='OPEN', XB=36.0,36.0,-3.81203,-3.270677,7.716418,8.313433/ 
&VENT ID='Mesh Vent: Mesh02-07-02 [XMIN]', SURF_ID='OPEN', XB=30.15,30.15,-3.81203,-3.270677,7.716418,8.313433/ 
&VENT ID='Inlet', SURF_ID='Supply', XB=32.365389,32.785389,-6.158866,-6.158866,7.776433,8.196433/ 

&SLCF QUANTITY='CELL V', VECTOR=.TRUE., ID='Mid_V', PBZ=7.985/
&SLCF QUANTITY='MIXTURE FRACTION', VECTOR=.TRUE., ID='Mid_MixtureFraction', PBZ=7.985/
&SLCF QUANTITY='TEMPERATURE', VECTOR=.TRUE., ID='Mid_Temp', PBZ=7.985/
&SLCF QUANTITY='RELATIVE HUMIDITY', VECTOR=.TRUE., ID='Mid_RH', PBZ=7.985/
&SLCF QUANTITY='TEMPERATURE', VECTOR=.TRUE., ID='Temp_X', PBX=32.575/
&SLCF QUANTITY='CELL V', VECTOR=.TRUE., ID='Vel_Y', PBX=32.575/

Thank you very much in advance

**Comment by drjfloyd** [comment]
The FDS Verification Guide has multiple examples on the evaporation of water droplets and all the inputs are provided to you in the Examples folder where you installed FDS.

**Reply by AlejandroEngESC** [reply]
In most of the evaporation examples that FDS provides water vapor is specified at the particle injection specie despite it is defined as Liquid in "particle type" and since no evaporation reaction is defined, some questions arise:
- Despite I define my particle as water vapor but select the particle type as Liquid, am I really injecting a liquid or directly the vapor? 
- Since no evaporation reaction is defined in most of the tutorials but the evaporation is reached anyways, this means that when selecting "water vapor" as the specie for the simulation it has already included the evaporation parameters?

I've tried to re-run the simulation using water vapor as the main injection specie instead of defining it, and evaporation is reached. When I post process the results I've selected water drops (the defined particle) to get results of MPUA at the walls and also the water vapor MPUA. Both results show the same distribution with the same values, therefore I'm wondering if there is any possibility to post-process the results to really see the real amount of water liquid and vapor that reach an specific boundary. 

Thank you in advance

**Reply by drjfloyd** [reply, marked as answer]
PART + SPEC_ID = liquid droplets

<img width="456" alt="image" src="https://github.com/user-attachments/assets/37731afd-fd9a-40cc-ad88-731d5a4897ac">

Table of predefined species indicating which species have predefined liquid properties:

![image](https://github.com/user-attachments/assets/121e413f-8163-4937-9568-c0e31904be4a)

MPUA is liquid only. Vapor is not attached to a surface:

<img width="442" alt="image" src="https://github.com/user-attachments/assets/56ee8a57-0f54-48d4-92a7-020b827cc72a">
